helper/helper/src/main/java/com/linkedin/avroutil1/compatibility/ConfigurableSchemaComparator.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright 2022 LinkedIn Corp.
+ * Licensed under the BSD 2-Clause License (the "License").
+ * See License in the project root for license information.
+ */
+
+package com.linkedin.avroutil1.compatibility;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import org.apache.avro.Schema;
+
+
+/**
+ * a more configurable alternative to {@link org.apache.avro.Schema#equals(Object)}
+ */
+public class ConfigurableSchemaComparator {
+
+  public static boolean equals(Schema a, Schema b, SchemaComparisonConfiguration config) {
+    validateConfig(config);
+    Set<SeenPair> seen = new HashSet<>(3);
+    return equals(a, b, config, seen);
+  }
+
+  private static void validateConfig(SchemaComparisonConfiguration config) {
+    if (config == null) {
+      throw new IllegalArgumentException("config required");
+    }
+    AvroVersion runtimeAvroVersion = AvroCompatibilityHelper.getRuntimeAvroVersion();
+    if (runtimeAvroVersion.earlierThan(AvroVersion.AVRO_1_7) && config.isCompareNonStringJsonProps()) {
+      //1.7 itself changes between < 1.7.3 and >= 1.7.3, so we leave that validation to later runtime :-(
+      throw new IllegalArgumentException("avro " + runtimeAvroVersion
+          + " does not preserve non-string props and so cannot compare them");
+    }
+  }
+
+  private static boolean equals(Schema a, Schema b,  SchemaComparisonConfiguration config, Set<SeenPair> seen) {
+    if (a == null && b == null) {
+      return true;
+    }
+    if (a == null || b == null) {
+      return false;
+    }
+    Schema.Type type = a.getType();
+    if (!Objects.equals(type, b.getType())) {
+      return false;
+    }
+
+    boolean considerJsonStringProps = config.isCompareStringJsonProps();
+    boolean considerJsonNonStringProps = config.isCompareNonStringJsonProps();
+    boolean considerAliases = config.isCompareAliases();
+
+    switch (type) {
+      //all of these have nothing more to compare by beyond their type (and we ignore props)
+      case NULL:
+        return true;
+      case BOOLEAN:
+        return true;
+      case INT:
+        return true;
+      case LONG:
+        return true;
+      case FLOAT:
+        return true;
+      case DOUBLE:
+        return true;
+      case STRING:
+        return true;
+      case BYTES:
+        return true;
+
+      //named types
+
+      case ENUM:
+        return a.getFullName().equals(b.getFullName())
+            && (!considerAliases || hasSameAliases(a, b))
+            && a.getEnumSymbols().equals(b.getEnumSymbols());
```

Comment:
What if the enum symbols are ordered differently? Are they still equal?

Reply:
no, and i think List.equals() covers us for that